ui/app/components/clients/activity.ts

```diff
@@ -0,0 +1,178 @@
+import { parseAPITimestamp } from 'core/utils/date-formatters';
+
+import type ClientsActivityModel from 'vault/models/clients/activity';
+import type ClientsVersionHistoryModel from 'vault/models/clients/version-history';
+
+interface Args {
+  activity: ClientsActivityModel;
+  versionHistory: ClientsVersionHistoryModel[];
+  startTimestamp: number;
+  endTimestamp: number;
+  currentTimestamp: number;
+  namespace: string;
+  authMount: string;
+}
+
+export default class ClientsActivityComponent extends Component<Args> {
+  get startTimeISO() {
+    return fromUnixTime(this.args.startTimestamp).toISOString();
+  }
+
+  get endTimeISO() {
+    return fromUnixTime(this.args.endTimestamp).toISOString();
+  }
+
+  get byMonthActivityData() {
+    const { activity, namespace } = this.args;
+    return namespace ? this.filteredActivityByMonth : activity.byMonth;
+  }
+
+  get filteredActivityByMonth() {
+    const { namespace, authMount, activity } = this.args;
+    if (!namespace && !authMount) {
+      return activity.byMonth;
+    }
+    const namespaceData = activity.byMonth
+      .map((m) => m.namespaces_by_key[namespace as keyof typeof m.namespaces_by_key])
+      .filter((d) => d !== undefined);
+
+    if (!authMount) {
+      return namespaceData.length === 0 ? null : namespaceData;
+    }
+
+    const mountData = authMount
+      ? namespaceData.map((namespace) => namespace?.mounts_by_key[authMount]).filter((d) => d !== undefined)
+      : namespaceData;
+
+    return mountData.length === 0 ? null : mountData;
+  }
+
+  get filteredActivityByNamespace() {
+    const { namespace, activity } = this.args;
+    return activity.byNamespace.find((ns) => ns.label === namespace);
+  }
+
+  get filteredActivityByAuthMount() {
+    return this.filteredActivityByNamespace?.mounts?.find((mount) => mount.label === this.args.authMount);
+  }
+
+  get filteredActivity() {
+    return this.args.authMount ? this.filteredActivityByAuthMount : this.filteredActivityByNamespace;
+  }
+
+  get isCurrentMonth() {
+    const { activity, currentTimestamp } = this.args;
+    const current = parseAPITimestamp(fromUnixTime(currentTimestamp)) as Date;
+    const start = parseAPITimestamp(activity.startTime) as Date;
+    const end = parseAPITimestamp(activity.endTime) as Date;
+    return isSameMonth(start, current) && isSameMonth(end, current);
+  }
+
+  get isDateRange() {
+    const { activity } = this.args;
+    return !isSameMonth(
+      parseAPITimestamp(activity.startTime) as Date,
+      parseAPITimestamp(activity.endTime) as Date
+    );
+  }
+
+  // (object) top level TOTAL client counts for given date range
+  get totalUsageCounts() {
+    const { namespace, activity } = this.args;
+    return namespace ? this.filteredActivity : activity.total;
+  }
+
+  get upgradeDuringActivity() {
+    const { versionHistory, activity } = this.args;
+    if (versionHistory) {
+      // filter for upgrade data of noteworthy upgrades (1.9 and/or 1.10)
```

Comment:
Do we want to make a ticket or add a todo to include 1.16 messaging?

Reply:
Yes good call I'll make a ticket and we'll circle back.